packages/framework/src/Component/Image/ImageLocator.php

```diff
@@ -58,11 +58,35 @@ public function getRelativeAdditionalImageFilepath(Image $image, int $additional
     {
         $path = $this->getRelativeImagePath($image->getEntityName(), $image->getType(), $sizeName);
 
-        $filename = str_replace(
-            ['{index}', '{filename}'],
-            [$additionalIndex, $image->getFilename()],
-            static::ADDITIONAL_IMAGE_MASK
-        );
+        $filename = $this->getAdditionalImageFilename($image->getFilename(), $additionalIndex);
+
+        return $path . $filename;
+    }
+
+    /**
+     * @param int $id
+     * @param string $extension
+     * @param string $entityName
+     * @param string|null $type
+     * @param string|null $sizeName
+     * @param int|null $additionalIndex
+     * @return string
+     */
+    public function getRelativeImageFilepathFromAttributes(
```

Comment:
As same problem as earlier - what are attributes? It seems to me they at least deserves a data object if they have a meaning...